drivers/amazonec2/amazonec2.go

```diff
@@ -566,12 +573,21 @@ func (d *Driver) Create() error {
 			DeleteOnTermination: aws.Bool(true),
 		},
 	}
-	netSpecs := []*ec2.InstanceNetworkInterfaceSpecification{{
-		DeviceIndex:              aws.Int64(0), // eth0
-		Groups:                   makePointerSlice(d.securityGroupIds()),
-		SubnetId:                 &d.SubnetId,
-		AssociatePublicIpAddress: aws.Bool(!d.PrivateIPOnly),
-	}}
+
+	var netSpecs []*ec2.InstanceNetworkInterfaceSpecification = nil
+	var securityGroupIds []*string = nil
+	var subnetId *string = nil
+	if !d.UseDefaultNetworkInterface {
+		netSpecs = []*ec2.InstanceNetworkInterfaceSpecification{{
```

Comment:
Shouldn't this be set to _something_ even if `UseDefaultNetworkInterface` is true?  Passing `nil` for this doesn't seem quite right to me